src/util-logopenfile.c

```diff
@@ -185,6 +187,15 @@ SCLogOpenFileFp(const char *path, const char *append_setting)
     if (ret == NULL)
         SCLogError(SC_ERR_FOPEN, "Error opening file: \"%s\": %s",
                    path, strerror(errno));
+
+    if (mode != 0) {
```

Comment:
I think we should not try to set the mode if the file failed to open.

Otherwise everything looks good and tests OK.